app/javascript/components/supplemental_application/sections/Lease.js

```diff
@@ -126,18 +127,45 @@ const Lease = ({ form, values, store }) => {
     ...map(confirmedPreferences, pluck('id', 'preference_name'))
   ])
 
-  const getVisited = (fieldName) => form.getFieldState(fieldName)?.visited
+  const availableUnits = units.filter(
+    (unit) => !unit.application_id || unit.application_id === application.id
+  )
 
-  const openDeleteLeaseConfirmation = () => setShowDeleteConfirmation(true)
-  const closeDeleteLeaseConfirmation = () => setShowDeleteConfirmation(false)
+  const availableUnitsOptions = formUtils.toOptions(
+    map(availableUnits, pluck('id', 'unit_number', 'priority_type'))
+  )
 
-  const handleDeleteLeaseConfirmed = () => {
-    setShowDeleteConfirmation(false)
-    handleDeleteLease()
-  }
+  const selectedUnit = form.getState().values.lease?.unit
 
   const areNoUnitsAvailable = !availableUnitsOptions.length
 
+  const prefUsedIs = (pref) => {
+    const prefUsed = confirmedPreferences.find(
+      (pref) => pref.id === form.getState().values.lease?.preference_used
+    )
+    return prefUsed?.preference_name.includes(pref)
+  }
+
+  const unitIsUsedWithPrefElsewhere = (unit, pref) =>
+    unit.application_id &&
+    unit.application_id !== application.id &&
+    unit.preference_used_name?.includes(pref)
```

Comment:
Can you explain the difference between preference_used_name on a unit and preference_used on a preference?

Reply:
preference_used_name is the name of the preference, vs preference_used on the lease is the application preference id, which is different for each application. I added the preference_used_name to the query instead of preference used because then we can figure out if it's DTHP or NRHP for any of the other applications.

Would a comment be helpful somewhere?

Reply:
No that's okay, not sure where the comment would go. Thanks for the explanation though!